kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -433,27 +456,122 @@ andEqualsFunctions
         , MonadUnify unifier
         )
     => [(SimplificationTarget, TermTransformation variable unifier)]
-andEqualsFunctions =
-    [ (AndT,    liftET boolAnd)
```

Comment:
Why not use a triple with the name, e.g.
```.hs
    [ (AndT, "boolAnd", liftET boolAnd)
    ...
```
Introducing a sum type for this list of functions will make every future change to this list non-local: adding a new case will require a new constructor above, a new case below, and an actual implementation elsewhere.

Actually, can't we get the necessary information for the log from `HasCallStack`?